### ISSUE TITLE
[FEATURE] Please implement an option to disable SSL certs validation

### ISSUE THREAD

**Issue by OlliHolli** [maximtrp/youtrack-vscode-extension#2]
For me the extension does not work.

My youtrack instance runs in a docker container with image `2021.4.38425`.  
I created an access token as described and even set the cors policy to unrestricted (allowAllOrigins), but the extension does not fetch my agiles.  
The error presented looks like this: `Error occurred while retrieving agiles`. I cannot see anything in the output window of vscode so I cannot provide more information.  

Furthermore I can fetch my agiles and issues using my access token when I use a program like postman without issues.

What could be the issue here?

**Comment by maximtrp** [comment]
Thank you for reporting! Please try the new version (v1.5.4) that includes more explicit error messages.

**Comment by OlliHolli** [comment]
Thanks for the quick update.  
The additional error info made debugging a thousand times better.  

Here the updated Error Info: 
```Failed to retrieve agiles, projects, and sprints. Error: unable to verify the first certificate```

This seems to me to be an issue with the self signed certificates our servers are using.  
Is there a possibility to allow self signed certificates in chain?

**Comment by maximtrp** [comment]
Yes, I can implement an option to bypass this check, but I would recommend obtaining Let's Encrypt certificates (there is a nice web server called Caddy that is doing it automatically) or maybe recreating your self-signed ones. If you are sure that nothing works for you except this option, I will add it (disabled by default).

**Comment by OlliHolli** [comment]
Well if this was a private situation I would happily use let's encrypt certificates.  
But since I'm in a company situation I cannot do that. I am forced to rely on the certificates I have, which are self signed for all internal affairs.  

That being said, I'd be more than happy if you spent the time and effort to implement such a bypass.

**Comment by cuda** [comment]
Hi, I have a similar issue using a PositiveSSL cert (not self-signed).  Is there any way to disable the cert check? Thanks

**Comment by kiber-io** [comment]
And I have such an error: `Failed to retrieve agiles, projects, and sprints. Error: timeout of 5000ms exceeded.`
Although the request goes through curl in a second or two. our server also uses its own certificates, maybe this affects

**Comment by maximtrp** [comment]
@kiber-io @OlliHolli @cuda Please try the new version

**Comment by OlliHolli** [comment]
First off, @maximtrp, thanks for taking the time to look into this.

I updated to version 1.6.0 of your extension and disabled the setting to verify certificates.  
Sadly your extension still reports the same issue `Failed to retrieve agiles, projects, and sprints. Error: unable to verify the first certificate`.

The terminal output in the `Window` section reads:

```shell
2024-10-21 07:48:54.642 [trace] CommandService#executeCommand youtrack.refreshAgiles
2024-10-21 07:48:54.643 [trace] MainThreadTreeViews#$refresh youtrack-agiles undefined
2024-10-21 07:48:54.644 [trace] MainThreadTreeViews#$refresh youtrack-agiles undefined
```

Is there more detailed error information in some output section I may have missed?

**Comment by maximtrp** [comment]
@OlliHolli Thank you for the fast response! Have you set the option in the settings to disable certification validation? Search by "youtrack" and you will see this checkbox.

**Comment by OlliHolli** [comment]
I did prior to my test today. 
As already mentioned this did not resolve my issue.

**Comment by maximtrp** [comment]
@OlliHolli This is unexpected. Unfortunately, I don't have such a server at hand, so I need to set it up to debug.

**Comment by maximtrp** [comment]
@OlliHolli Could you please test the new version?

**Comment by OlliHolli** [comment]
@maximtrp, thanks again for taking the time for this.
I can confirm I was now able to retrieve Issues, Agiles and the like from our Youtrack server! 💯 

I see this issue / feature as implemented and closed.
Thanks for the nice work.
